File: src/Sample/Flutter.Net/CustomSliversDemoPage.cs
using Avalonia;
using Avalonia.Media;
using Flutter.Rendering;
using Flutter.Widgets;

// Dart parity source (reference): dart_sample/lib/custom_slivers_demo_page.dart (exact sample parity)

namespace Flutter.Net;

public sealed class CustomSliversDemoPage : StatelessWidget
{
    public override Widget Build(BuildContext context)
    {
        return new CustomScrollView(
            slivers:
            [
                new SliverToBoxAdapter(
                    new Container(
                        color: Colors.White,
                        padding: new Thickness(12),
                        child: new Column(
                            crossAxisAlignment: CrossAxisAlignment.Stretch,
                            spacing: 6,
                            children:
                            [
                                new Text("CustomScrollView + Slivers", fontSize: 20, color: Colors.Black),
                                new Text(
                                    "SliverPadding and SliverFixedExtentList are used directly.",
                                    fontSize: 14,
                                    color: Colors.DimGray),
                            ]))),
                new SliverPadding(
                    padding: new Thickness(12, 10, 12, 8),
                    sliver: SliverFixedExtentList.Builder(
                        childCount: 18,
                        itemExtent: 42,
                        itemBuilder: (_, index) => new Container(
                            color: index % 2 == 0 ? Color.Parse("#FFE8F5E9") : Color.Parse("#FFEAF4FF"),
                            padding: new Thickness(10, 8),
                            child: new Text(
                                $"fixed sliver row #{index}",
                                fontSize: 13,
                                color: Colors.Black)),
                        addAutomaticKeepAlives: false)),
                new SliverPadding(
                    padding: new Thickness(12, 8, 12, 16),
                    sliver: SliverList.Builder(
                        childCount: 8,
                        itemBuilder: (_, index) => new Container(
                            color: Color.Parse("#FFF5F5F5"),
                            padding: new Thickness(10, 10),
                            child: new Text(
                                $"regular sliver row #{index}",
                                fontSize: 13,
                                color: Colors.Black)),
                        addAutomaticKeepAlives: false)),
            ]);
    }
}

File: dart_sample/lib/custom_slivers_demo_page.dart
import 'package:flutter/material.dart';

class CustomSliversDemoPage extends StatelessWidget {
  const CustomSliversDemoPage({super.key});

  @override
  Widget build(BuildContext context) {
    return CustomScrollView(
      slivers: <Widget>[
        const SliverToBoxAdapter(
          child: ColoredBox(
            color: Colors.white,
            child: Padding(
              padding: EdgeInsets.all(12),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.stretch,
                spacing: 6,
                children: <Widget>[
                  Text(
                    'CustomScrollView + Slivers',
                    style: TextStyle(fontSize: 20, color: Colors.black),
                  ),
                  Text(
                    'SliverPadding and SliverFixedExtentList are used directly.',
                    style: TextStyle(fontSize: 14, color: Colors.black54),
                  ),
                ],
              ),
            ),
          ),
        ),
        SliverPadding(
          padding: const EdgeInsets.fromLTRB(12, 10, 12, 8),
          sliver: SliverFixedExtentList.builder(
            itemExtent: 42,
            itemCount: 18,
            itemBuilder: (BuildContext context, int index) {
              return Container(
                color: index.isEven
                    ? const Color(0xFFE8F5E9)
                    : const Color(0xFFEAF4FF),
                padding: const EdgeInsets.symmetric(
                  horizontal: 10,
                  vertical: 8,
                ),
                child: Text(
                  'fixed sliver row #$index',
                  style: const TextStyle(fontSize: 13, color: Colors.black),
                ),
              );
            },
          ),
        ),
        SliverPadding(
          padding: const EdgeInsets.fromLTRB(12, 8, 12, 16),
          sliver: SliverList.builder(
            itemCount: 8,
            itemBuilder: (BuildContext context, int index) {
              return Container(
                color: const Color(0xFFF5F5F5),
                padding: const EdgeInsets.symmetric(
                  horizontal: 10,
                  vertical: 10,
                ),
                child: Text(
                  'regular sliver row #$index',
                  style: const TextStyle(fontSize: 13, color: Colors.black),
                ),
              );
            },
          ),
        ),
      ],
    );
  }
}
